src/Enum.php

```diff
@@ -199,11 +205,13 @@ public static function isValid($value)
      * @psalm-pure
      * @psalm-assert T $value
      */
-    public static function assertValidValue($value): void
+    private static function assertValidValue($value): string
```

Comment:
I propose to make private the newly public static method added in #135 as there was no new release so far.
- If a user wants to programmatically test that a value is valid, it can do this by using `isValid()` static method.
- During instantiation, `from()` can be used to have an exception thrown for invalid values.
- I want to easily re-use the method to return the key when a value is valid for further storing in constructor.

Reply:
@michaelpetri, let me know if this makes sense and if you envision any other use-case that I missed

Reply:
I worked around it for now.

Reply:
Sorry for my late response. imo no need to have it public since i just can call `from($mixed)` and get an enum or exception.

Reply:
Sorry missed the return string part, imo the assert should not return anything. If no exception was thrown you know you have a valid key, or do i miss something?

Reply:
The current version of the PR is with an additional private method called `assertValidValueReturningKey()` that returns string and having the current `assertValidValue()` still there calling it.

The operation does a full array scan and I would like to reuse the same array scan for retrieving the key in one operation, changing `in_array` with `array_search`.

`assertValidValueReturningKey()` would be called in ctor and in `from()` method as well.

Reply:
Okay, now I understand what you want to achieve. From my side it's ok, even if I think it makes the codebase a bit more confusing. Would another static map (value => key) be an option here as well? Then you could lookup keys by `self::$otherMap[$value] ?? null` ?

Reply:
That would not work as values can be anything, not just int or string.

Reply:
Okay, so nothing more from my side. :zipper_mouth_face: